src/pybind/mgr/dashboard/frontend/src/app/core/auth/role-form/role-form.component.ts

```diff
@@ -0,0 +1,144 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+
+import { BsModalRef } from 'ngx-bootstrap';
+
+import { RoleService } from '../../../shared/api/role.service';
+import { ScopeService } from '../../../shared/api/scope.service';
+import { NotificationType } from '../../../shared/enum/notification-type.enum';
+import { CdFormGroup } from '../../../shared/forms/cd-form-group';
+import { NotificationService } from '../../../shared/services/notification.service';
+import { RoleFormMode } from './role-form-mode.enum';
+import { RoleFormModel } from './role-form.model';
+
+@Component({
+  selector: 'cd-role-form',
+  templateUrl: './role-form.component.html',
+  styleUrls: ['./role-form.component.scss']
+})
+export class RoleFormComponent implements OnInit {
+  modalRef: BsModalRef;
+
+  roleForm: CdFormGroup;
+  response: RoleFormModel;
+  scopes: Array<string>;
+
+  roleFormMode = RoleFormMode;
+  mode: RoleFormMode;
+
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+    private roleService: RoleService,
+    private scopeService: ScopeService,
+    private notificationService: NotificationService
+  ) {
+    this.createForm();
+  }
+
+  createForm() {
+    this.roleForm = new CdFormGroup({
+      name: new FormControl('', {
```

Comment:
Consider to use the CdValidators.unique in the role add formular to check if the role name already exists. See https://github.com/ceph/ceph/pull/23802 and https://github.com/votdev/ceph/blob/631a9c9844225ec47bb175a893da39795ec50e7b/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-form/rgw-user-form.component.ts#L54 how to implement it.